datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/DurationDeserializer.java

```diff
@@ -52,6 +58,13 @@ public Duration deserialize(JsonParser parser, DeserializationContext context) t
         {
             case JsonTokenId.ID_NUMBER_FLOAT:
                 BigDecimal value = parser.getDecimalValue();
+                // If the decimal isnt within the bounds of a float, bail out
+                if(value.compareTo(INSTANT_MAX) > 0 ||
+                        value.compareTo(INSTANT_MIN) < 0) {
+                    throw new JsonParseException(context.getParser(),
+                            "Value of Float too large to be converted to Duration");
```

Comment:
Turns out this isn't the correct boundary: `Duration` can hold a number-of-seconds up to `Long.MAX_VALUE`, which is larger than `Instant.MAX.getEpochSecond()`.

For larger values, Jackson currently has the troubling behavior of `BigDecimal.longValue()`.  This test passes:
```
    @Test
    public void testDeserializationAsFloat05() throws Exception
    {
        String customInstant = new BigInteger(Long.toString(Long.MAX_VALUE)).add(BigInteger.ONE) + ".0";
        Duration value = READER.without(DeserializationFeature.READ_DATE_TIMESTAMPS_AS_NANOSECONDS)
                                 .readValue(customInstant);
        assertEquals(Long.MIN_VALUE, value.getSeconds());   // The Duration is negative!
    }
```

Reply:
For reference on behavior of similar overflow, `Duration.of(Long.MAX_VALUE, ChronoUnit.MINUTES)` throws:
```
java.lang.ArithmeticException: long overflow
```

while `Duration.parse("PT" + Long.MAX_VALUE + "0S")` -- note the added `0` to force overflow -- throws:
```
java.time.format.DateTimeParseException: Text cannot be parsed to a Duration: seconds
```